soap/src/java/com/zimbra/soap/mail/message/TestDataSourceResponse.java

```diff
@@ -17,53 +17,62 @@
 
 package com.zimbra.soap.mail.message;
 
-import com.google.common.base.Objects;
+import java.util.Collections;
+import java.util.List;
+
 import javax.xml.bind.annotation.XmlAccessType;
 import javax.xml.bind.annotation.XmlAccessorType;
-import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlElements;
 import javax.xml.bind.annotation.XmlRootElement;
 
+import com.google.common.base.Objects;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
 import com.zimbra.common.soap.MailConstants;
-import com.zimbra.soap.type.ZmBoolean;
+import com.zimbra.soap.mail.type.TestDataSource;
 
 @XmlAccessorType(XmlAccessType.NONE)
 @XmlRootElement(name=MailConstants.E_TEST_DATA_SOURCE_RESPONSE)
 public class TestDataSourceResponse {
-
     /**
-     * @zm-api-field-tag success
-     * @zm-api-field-description Flags whether the test was successful
+     * @zm-api-field-description Data source information
      */
-    @XmlAttribute(name=MailConstants.A_DS_SUCCESS /* success */, required=true)
-    private final ZmBoolean success;
+    @XmlElements({
+        @XmlElement(name=MailConstants.E_DS_IMAP /* imap */, type=TestDataSource.class),
+        @XmlElement(name=MailConstants.E_DS_POP3 /* pop3 */, type=TestDataSource.class),
+        @XmlElement(name=MailConstants.E_DS_CALDAV /* caldav */, type=TestDataSource.class),
+        @XmlElement(name=MailConstants.E_DS_YAB /* yab */, type=TestDataSource.class),
+        @XmlElement(name=MailConstants.E_DS_RSS /* rss */, type=TestDataSource.class),
+        @XmlElement(name=MailConstants.E_DS_GAL /* gal */, type=TestDataSource.class),
+        @XmlElement(name=MailConstants.E_DS_CAL /* cal */, type=TestDataSource.class),
+        @XmlElement(name=MailConstants.E_DS_OAUTH /* oauth */, type=TestDataSource.class),
+        @XmlElement(name=MailConstants.E_DS_UNKNOWN /* unknown */, type=TestDataSource.class)
+    })
+    private List<TestDataSource> dataSources = Lists.newArrayList();
 
-    /**
-     * @zm-api-field-tag error-message
-     * @zm-api-field-description Error message
-     */
-    @XmlAttribute(name=MailConstants.A_DS_ERROR /* error */, required=false)
-    private String error;
+    public void setDataSources(Iterable <TestDataSource> dataSources) {
+        this.dataSources.clear();
+        if (dataSources != null) {
+            Iterables.addAll(this.dataSources,dataSources);
+        }
+    }
 
-    /**
-     * no-argument constructor wanted by JAXB
-     */
-    @SuppressWarnings("unused")
     private TestDataSourceResponse() {
-        this(false);
     }
 
-    public TestDataSourceResponse(boolean success) {
-        this.success = ZmBoolean.fromBool(success);
+    public TestDataSourceResponse addDataSource(TestDataSource dataSource) {
+        this.dataSources.add(dataSource);
+        return this;
     }
 
-    public void setError(String error) { this.error = error; }
```

Comment:
Isn't this an API change?  Will no longer have `error` and `success`

Reply:
no, it's a fix to the JAXB to make JAXB match the API. The API already behaves differently (and probably always had), but the JAXB class does not correspond to how API actually behaves.

Reply:
Ok. That makes sense.